### ISSUE TITLE
About Seg Embedding

### ISSUE THREAD

**Issue by qhd1996** [microsoft/vert-papers#3]
According to the paper, BMES scheme(Convolutional Neural Network with Word Embeddings for
Chinese Word Segmentation) is used to represent CWS. Only four types of tags is enough to label all characters.
My question is, in your code, there are five tags(OBMES) in soft seg embedding, why?

**Comment by topDreamer** [comment]
和楼上遇到同样的问题麻烦解答。
另外，还遇到两个问题，第一个是想咨询下数据格式是怎么样的，能否给下样例数据，因为看数据处理的过程，看不到seg信息是如何添加的；
第二个问题是，在nner_with_seg_info.py文件中定义了biGRU，但是好像没有在模型中使用，请问是为什么？

**Comment by wgx998877** [comment]
>My question is, in your code, there are five tags(OBMES) in soft seg embedding, why?

For real production, we need to handle the input with missing  segmentation in same model structure, the "O" is used to represent on such case.

**Comment by wgx998877** [comment]
>另外，还遇到两个问题，第一个是想咨询下数据格式是怎么样的，能否给下样例数据，因为看数据处理的过程，看不到seg信息是如何添加的；
第二个问题是，在nner_with_seg_info.py文件中定义了biGRU，但是好像没有在模型中使用，请问是为什么？

1) there are some license issues to prevent to share those raw dataset directly from Microsoft.
But you might reference some existing papers from web, for example, in Weibo dataset:
https://github.com/OYE93/Chinese-NLP-Corpus/blob/master/NER/Weibo/weiboNER_2nd_conll.train
you can find the segmentation information from their dataset(number after word):
科0	O技1	O全0	O方1	O位2	O资0	O
So, it would be easy to convert such segmentation index to BMES label schema and pass it to model embeddings.
For other datasets, the paper is using existing tool/method to generate the segmentations. And you can also use other open source tools for this purpose.

2) it's clear a missing line to use this as running ablation studies.
something should be missing before line 72 in like `cnn_out = self.rnn(cnn_out)`
https://github.com/microsoft/vert-papers/blob/master/papers/CAN-NER/model/nner_with_seg_info.py
